### PR TITLE
Feature/e2e testing

### PR DESCRIPTION
**Pull request by ikaem**

### CODE REVIEW

File: src/tests/api.test.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      createdAt: '2021-01-21T09:59:15.465864',
+      updatedAt: '2021-01-21T09:59:15.465864',
+      category: { id: 2854, slug: 'sport', name: 'sport', itemCount: null },
+      data: {
+        author: '',
+        title: 'Football',
+        description: '',
+        text: '',
+        whatIsIt: '',
+        modelNumber: '',
```
Comment (tomislavpet) [inline src/tests/api.test.ts]: can we please also add these fields and check what is returned when we send `lang=en` and `lang=hr` and `lang=it` parameters?

File: src/tests/setup.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import SiteSails from '../index';
+
+const ss = new SiteSails({
+  apiUrl: 'https://api.sitesails.com/api/v1',
+  // TODOD admin COPIED FROM CGH
+  adminApiKey: '0fe40734-ae13-4eee-9782-df342c203bda',
+  publicApiKey: '66461de9-b081-43ed-87c4-7fa098829251',
+});
+
+export const fetchNodes = async (params: NodeSearchParams = {}) => {
```
Comment (tomislavpet) [inline src/tests/setup.ts]: we probably don't need these helper wrappers but can simply export `ss` from here and use it directly in tests - these wrappers aren't adding anything other than re-calling the same methods

File: src/tests/api.test.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    expect(data).toHaveLength(4);
+  });
+
+  it('should be fetched only from a specified node category', async () => {
+    const category = 'sport';
+    const { data } = await fetchNodes({ category });
+
+    expect(data).toHaveLength(2);
+  });
```
Comment (vkozinec) [inline src/tests/api.test.ts]: add space

File: src/tests/api.test.ts [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+    const { data } = await ss.nodes(nodes).get(productSlug, { lang });
+
+    expect(data.title).toEqual('Nogometna lopta');
+  });
+
+  it('should be of specific shape when a node is fetched', async () => {
+    const expectedProduct = {
+      id: 2856,
+      slug: 'football',
+      // TODO is slugs added recently?
```
Comment (tomislavpet) [inline src/tests/api.test.ts]: yes, added a day ago and should return slugs for all translated languages. I.e.:

```
{
  'en': 'english-slug',
  'hr': 'hrvatski-slug'
}
```

File: src/tests/api.test.ts [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+    };
+    const lang = 'hr';
+
+    const { data } = await ss.nodes(nodes).get(productSlug, { lang });
+
+    expect(data).toEqual(expectedResponse);
+  });
+
+  it('should fetch a node with Italian translations when Italian language specified', async () => {
+    const expectedResponse = {
```
Comment (tomislavpet) [inline src/tests/api.test.ts]: the idea here is to not have Italian translation in SS and have it fallback to default language (english)

File: src/types.ts [hunk 6]
```diff
@@ -26,6 +26,8 @@ export type Node = {
   connections: any;
   connectionStats: any;
   event: NodeEvent;
+  // TODO ADDED LANGUAGE ID? TO BE ABLE TO ACCESS LANGUAGE IN THE TESTS
```
Comment (tomislavpet) [inline src/types.ts]: it's added to the latest version, rebase on master

File: src/tests/api.test.ts [hunk 7]
```diff
@@ -219,6 +219,10 @@ describe('Api members', () => {
   // });
   // console.log('res', res);
   // });
+<<<<<<< HEAD
+
+=======
+>>>>>>> 9697799310f22736934e1ef30254cce1c7c00fe0
```
Comment (tomislavpet) [inline src/tests/api.test.ts]: there are merge artifacts in multiple places - that should be all cleaned up an resubmitted

**Pull request merged**